app/actions/v3/service_instance_update_managed.rb

```diff
@@ -0,0 +1,362 @@
+require 'services/service_brokers/service_client_provider'
+require 'actions/metadata_update'
+require 'cloud_controller/errors/api_error'
+
+module VCAP::CloudController
+  module V3
+    class ServiceInstanceUpdateManaged
+      class UnprocessableUpdate < CloudController::Errors::ApiError; end
+      class InvalidServiceInstance < StandardError
+      end
+      class LastOperationFailedState < StandardError; end
+
+      PollingStatus = Struct.new(:finished, :retry_after).freeze
+      PollingFinished = PollingStatus.new(true, nil).freeze
+      ContinuePolling = ->(retry_after) { PollingStatus.new(false, retry_after) }
+
+      UPDATE_IN_PROGRESS_OPERATION = { type: 'update', state: 'in progress' }.freeze
+
+      def initialize(instance, message, user_audit_info, audit_hash)
+        @service_instance = instance
+        @message = message
+        @user_audit_info = user_audit_info
+        @audit_hash = audit_hash
+      end
+
+      def precursor
+        raise_if_cannot_proceed!
+
+        unless is_deleting?(service_instance) || !only_metadata?
```

Comment:
This method is hard to reason about. It takes a lot of effort to work out what `unless !only_metadata?` means, and why the metadata could be updated in more than one place.

Reply:
Updated the if statement to `if update_metadata_only?`. Hope this makes it more clear